test/service/test_request_validation.py

```diff
@@ -177,3 +179,43 @@ def test_manifest(self):
             manifest_config = json.load(open('{}/request_config.json'.format(self.service_config_dir), 'r'))['manifest']
             expected_fieldnames = list(manifest_config['bundles'].keys()) + list(manifest_config['contents.files'].keys())
             self.assertEqual(expected_fieldnames, tsv_file.fieldnames, 'Manifest headers are not configured correctly')
+
+    @mock_sts
+    @mock_s3
+    def test_bdbag_manifest(self):
+        logging.getLogger('test_request_validation').\
+            warning('test_manifest is invoked')
+        # moto will mock the requests.get call so we can't hit localhost;
+        # add_passthru let's us hit the server
+        # see this GitHub issue and comment:
+        # https://github.com/spulec/moto/issues/1026#issuecomment-380054270
+        with ResponsesHelper() as helper, TemporaryDirectory() as zip_dir:
+            helper.add_passthru(self.base_url)
+            storage_service = StorageService()
+            storage_service.create_bucket()
+            url = self.base_url +\
+                  '/repository/files/export?filters={"file":{}}&format=bdbag'
+            response = requests.get(url)
+            self.assertEqual(200, response.status_code,
+                             'Unable to download manifest')
+            zip_fh = ZipFile(response.text, 'r')
+            zip_fh.extractall(zip_dir)
+            zip_fh.close()
+            zip_fname = os.path.basename(os.path.splitext(response.text)[0])
+            fh = open(os.path.join(zip_dir, zip_fname, 'data', 'sample.tsv'))
+            tsv_file = csv.DictReader(fh, delimiter='\t')
+            # 2 because self.bundle has 2 files
+            self.assertEqual(len(list(tsv_file)), 2,
+                             'Wrong number of files were found.')
+            manifest_config = json.load(
+                open('{}/request_config.json'.format(self.service_config_dir),
+                     'r'))['manifest']
+            expected_fieldnames = \
```

Comment:
Don't make this dynamic, `request_config.json` is going away. Just hard code the expected column names.

Reply:
The problem with this style of dynamically generated assertion is that it uses exactly the same approach as the code under test. If there is bug in that approach, the assertion will also be wrong and the bug will go undetected.

Reply:
fixed